WebDevStudios/wd_s
# Add editor styles entry point

I don't think this closes any issue, but it's related to a couple of things noted in #614.

### DESCRIPTION
Adds a new entry point for editor styles in `/src/editor/editor.js`.

This also updates our npm scripts from using our `wds-build` and `wds-start` scripts to `wp-scripts` so that we can easily modify our entry points:

```
    "build": "wp-scripts build src/index.js src/editor/editor.js && npm run theme",
    "build:editor": "wp-scripts build src/editor/editor.js",
    "start": "wp-scripts start src/index.js src/editor/editor.js",
```

This *does* increase compiling times when running `npm run start` on my machine from around 2600ms on `main` to around `5000ms` in this branch. It'd be great to reduce that number as much as possible, of course, if we can.

Since editor styles are added through theme support hooks, we don't need to worry about wrapping our Tailwind or theme styles in a tag to scope them – this happens naturally:
![editor-styles-wrapper class wrapping our styles](https://d.pr/i/KYNNfp+)

### SCREENSHOTS
![terminal screenshot](https://d.pr/i/x4aGHu+)
![updated header styles](https://d.pr/i/2ibPST+)
![header styles on the frontend](https://d.pr/i/XmvJpy+)
![header styles on the backend](https://d.pr/i/ngLs8c+)

### OTHER


### STEPS TO VERIFY

1. Checkout the branch
2. Run `npm install`
3. Run `npm run build` to do a full build and check your editor and frontend styles
4. Run `npm run build:editor` to ONLY compile editor styles. **NOTE:** this ONLY compiles editor styles and NOT styles on the frontend. We may not even need this, but it is useful for testing so if we decide to go this route we could keep it in until we want to merge then yank it out.
5. Run `npm run start` and/or `npm run start:sync` to start the watch process and make some changes, then check the editor and frontend. Compile times did not differ for me running the basic `start` versus the Browsersync `start:sync`.

### DOCUMENTATION
I think probably, yes – we'll want to mention the update to our npm scripts and how we're including editor styles.

Assigning to @gregrickaby for review but also cc @michealengland to test out and compare to his work in another branch. We may be able to combine some thoughts around this to get a more efficient process running.
